### PR TITLE
feat(automation): add a script to automatically generate release branches

### PR DESCRIPTION
**Pull request by willcumbie**
Automates the creation of a new release pull request. Follow-up pull request will add a second script to call the GitHub API and create the release with relevant attachments.

### CODE REVIEW

**Comment by Jigarsolanki** [comment]
👎 it should be in node.js :) 

LGTM

**Comment by willcumbie** [comment]
Actually thought seriously about trying to do this in Node, but it turns out I hate Node. It'd probably also work as a shell script, though the file manipulation might be a little harder to read.

**Comment by adampickeral** [comment]
it's cool, but i'd prefer to not add python as a dependency to this project

File: scripts/create_release_branch.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+changelog_contents = changelog.readlines()
+changelog.close()
+new_changelog_contents = ['### Version ' + new_version + '\n'] + ['\n'.join(change_lines), '\n\n'] + changelog_contents
+changelog = open(os.path.dirname(__file__) + '/../CHANGELOG.md', "w")
+new_changelog_contents = "".join(new_changelog_contents)
+changelog.write(new_changelog_contents)
+changelog.close()
+
+# commit the branch and push it
+formatted_change_list = '\n'.join(change_lines).replace('"','')
```
Comment (akrikos) [inline scripts/create_release_branch.py]: You may want to warn folks that any double quotes will be removed from the change log on line 31

**Comment by akrikos** [comment]
Other than the nit about warning the user about double quotes (assuming I'm reading that correctly), it looks good.

**Comment by willcumbie** [comment]
@adampickeral I can rewrite this to a shell script, sure, but I feel like that's going to be a poor choice, if not completely unworkable, for the next script, which will need to make API calls and process JSON data. Python is pretty ubiquitous, and it's an optional dependency, no?

**Comment by adampickeral** [comment]
hmm, ok I'll go with it

**Pull request merged**